### ISSUE TITLE
ImageAnimatorWCS unit_x_axis and unit_y_axis do not follow image_axes

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#2855]
The units are always applied to the last two dimensions of the WCS, which is just wrong.

**Comment by Naman9639** [comment]
I would like to work on this issue

**Comment by nabobalis** [comment]
So the issue seems to be that [here](https://github.com/sunpy/sunpy/blob/master/sunpy/visualization/animator/image.py#L228
) we have hard coded that `axes.coords[2]` and `axes.coords[1]`, we assume that `x` and `y` are those two indexes, but if a user sets `image_axes=[-1, -2] `, i.e., in opposite order then it's wrong.

I'm not sure how it should be fixed as of this point.

**Comment by Naman9639** [comment]
Can we not restrict the user from setting those values?

**Comment by nabobalis** [comment]
You mean `image_axes=[-1, -2] `? Unfortunately not.

**Comment by nabobalis** [comment]
I'm not sure I follow.

**Comment by nabobalis** [comment]
I am not aware that there is a formal example that does that within SunPy. (We should add one).

But there is this comment from an open pull request: https://github.com/sunpy/sunpy/pull/2566#issuecomment-379249252

**Comment by Naman9639** [comment]
So for now I was thinking that if the user enters image_axes=[-1,-2] then we can simply put an if statement in the init function which just reverse the image_axes list. In this way even if the user enters the image_axes in reverse order it won't give wrong answer

**Comment by nabobalis** [comment]
I believe that the issue is that `ImageAnimatorWCS` allows n-dimensional input. So they could enter any `image_axes`.

**Comment by Naman9639** [comment]
Ahh ok I'll see what I can do

**Comment by Naman9639** [comment]
I am not sure how to approach it.

**Comment by nabobalis** [comment]
I'm not sure there is a straightforward answer. It would probably have to check the WCS information along side image_axis.

**Comment by Naman9639** [comment]
I'll try all that I can do.

**Comment by nabobalis** [comment]
Yeah, it does seem that `self.slices_wcsaxes` has that information. So that could work. But I think we need to come up with a bugged example to check that the fix works.
